src/Engine/Army.java

```diff
@@ -265,50 +292,43 @@ protected boolean awbwDeniesCharge()
   @Override
   public GameEventQueue receiveBattleEvent(final BattleSummary summary)
   {
-    if( awbwDeniesCharge() )
-      return null;
+    // The lambdas demand the args be effectively final, so we're duplicating checks today
```

Comment:
And why, oh why was your solution to use a lambda in the first place instead of simply allowing the `*Charge` function to accept `(Commander co, UnitDelta minion, UnitDelta enemy, boolean isCounter)` as parameters and calling `co.calculateCombatCharge(minion, enemy, isCounter)` inside the functions?

It adds indirection and complexity, all for the sake of calling a function that is readily available already without wrapping it and passing it around.

Reply:
Ah, okay. Should have read farther before complaining. Makes more sense since the underlying calculation can change between battles and mass damage.
So while I still don't love lambdas as much as you, I get why you are using one here.